### ISSUE TITLE
Sponsoring Package Phobia

### ISSUE THREAD

**Issue by styfle** [styfle/packagephobia#461]
There have been a few requests to increase the amount of data the Package Phobia is tracking.

- #124 add file count and directory count
- #442 add publish date
- #141 show badge on every cnpm package

Google Analytics shows the current monthly unique users is about 2000 but I suspect this actually much higher considering ad blockers will prevent GA from collecting stats and many developers use ad blockers.

Vercel currently sponsors the compute but I pay for the database out of pocket (about $85/year). The next tier up with redislabs would be about $170/year.

I would be willing to switch to a different database if a DBaaS wanted to sponsor this project. Otherwise, I will probably apply for the GitHub Sponsors program.

Put a 👍 if you would be willing to donate to this project so I can see if this upgrade is feasible.

**Comment by XhmikosR** [comment]
You should add GitHub Sponsors regardless IMO. Keep the funds for a rainy day or something. This is a too valuable project IMO.

I just think most developers don't even consider a package's size... That being said you are definitely right about ad blockers and tracking protection, so the stats are probably off.

Another solution would be to check with @coderberry to see if you can get codefund for the site? I don't like ads either, but this is different than the traditional ads.

**Comment by styfle** [comment]
I applied to GitHub Sponsors and should hear back in a week or two, thanks!

Years ago I was using Code Sponsor, now rebranded as Code Fund. But they changed their requirements to be much more strict. I'll definitely need to get a more accurate impressions count before trying to apply for that one again. We'll see if GH Sponsors can meet the funding goals and maybe ads won't be necessary.

**Comment by coderberry** [comment]
> You should add GitHub Sponsors regardless IMO. Keep the funds for a rainy day or something. This is a too valuable project IMO.
> 
> I just think most developers don't even consider a package's size... That being said you are definitely right about ad blockers and tracking protection, so the stats are probably off.
> 
> Another solution would be to check with @coderberry to see if you can get codefund for the site? I don't like ads either, but this is different than the traditional ads.

We'd love to help out! Please send over a request via codefund.io and we'll get you set up!

**Comment by styfle** [comment]
Thanks @coderberry, I just submitted a request 👍

**Comment by styfle** [comment]
@coderberry I was able to successfully register and I added the script to the home page in PR #478. 

However, I still see the status is _Pending_ and when I hover over the yellow icon says "Awaiting Administrator Approval".

I also noticed that the JS + CSS payload is quite large:

- 32 KB for JS
- 38 KB for CSS

Page load time went from 330ms to 850ms.

**Comment by XhmikosR** [comment]
Yeah, unfortunately it does affect page load time a lot. The CSS is a recent addition, which IMHO can be removed. That being said, it still is slow and most importantly it doesn't seem to use any compression @coderberry . Also, I was under the impression that Cloudflare was being used everywhere but apparently it's not?

https://gtmetrix.com/compare/uXsc5wN1/OY8DvQBq

**Comment by coderberry** [comment]
@styfle @XhmikosR we are aware of the speed issues. It's something that is actively being worked on. Hopefully we will see speed increases over the next day or two. Please feel free to assist in this as well if interested.

**Comment by coderberry** [comment]
@styfle the property is now approved. Sorry for the delay.

**Comment by styfle** [comment]
It's been over a year and a lot has changed. Here's a quick summary:

1. Thanks to @coderberry, I started using CodeFund to serve relevant ads. Unfortunately, they went out of business.
2. So I switched to Carbon Ads, but the payout was less than $1 per month.
3. So I switched to Ethical Ads making about $5 per month. However, I need to make $50 before I can cash out and the current balance is only $30.
4. GitHub Sponsors has found me between 1 and 3 sponsors each at $1 per month, for a total of $20! And I confirmed this does payout as low as $1 a month, so it doesn't seem there is a minimum which is really nice.
5. All that said, I am still not able to break even which leads me to a new chapter....

Thanks to @noahfschr for sponsoring the redis database, I have now switched from redislabs to upstash. This means I can finally cover the cost of hosting (vercel.com) and data storage (upstash.com) 100% without paying out of pocket! 🎉 💯
